lib/versioning/VersionID.js

```diff
@@ -240,14 +260,19 @@ function encode(str, encType = ENC_TYPE_HEX) {
  * @return {(string|Error)} - the decoded versionId or an error
  */
 function decode(str) {
-    if (str.length < 40) {
+    // default format is exactly 32 characters when encoded
+    if (str.length === 32) {
         return base62Decode(str);
```

Comment:
I'd suggest adding an assert to check the decoded size, because for such an important invariant, in the unlikely case it would be broken at some point it seems better to crash than to store invalid version IDs with the risk of corruption:

```suggestion
        const decoded = base62Decode(str);
        assert.strictEqual(decoded.length, 27);
        return decoded;
```